### PR TITLE
Remove non-state transforms

### PR DESCRIPTION
**Pull request by rjernst**
Reserved cluster state is build from handlers that modify cluster state. In the past there was one "non-state" handler that stored it's state outside cluster state. However, with that use case now gone, non-state transforms are no longer needed. This commit removes support for non-state transforms.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/reservedstate/TransformState.java [hunk 1]
```diff
@@ -24,8 +22,4 @@
  * other state outside the cluster state. The consumer, if provided, must return a {@link NonStateTransformResult} with
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/reservedstate/TransformState.java:22]: Reference to NonStateTransformResult here

File: server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java [hunk 2]
```diff
@@ -453,5 +370,5 @@ public void installStateHandler(ReservedClusterStateHandler<?> handler) {
     /**
      * Helper record class to combine the result of a trial run, non cluster state actions and any errors
      */
-    record TrialRunResult(List<Consumer<ActionListener<NonStateTransformResult>>> nonStateTransforms, List<String> errors) {}
+    record TrialRunResult(List<String> errors) {}
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java:373]: Do we still need a separate type here?

**Comment by thecoop** [comment]
Is this PR meant to remove lots of ES|QL types too?

File: server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java (hunk 2 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java]: Removed

File: server/src/main/java/org/elasticsearch/reservedstate/TransformState.java (hunk 1 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/reservedstate/TransformState.java]: Removed

**Comment by rjernst** [comment]
@thecoop This should be clean now

**Review by thecoop — APPROVED** [review]

**Pull request merged**